appIcons.js

```diff
@@ -67,9 +67,11 @@ const MyAppIcon = new Lang.Class({
     Extends: AppDisplay.AppIcon,
 
     // settings are required inside.
-    _init: function(settings, app, iconParams, onActivateOverride) {
+    _init: function(settings, app, monitorNumber, iconParams, onActivateOverride) {
```

Comment:
Sure thing, I'll rename the variables!

As for the "bigger thing", it's not clear to me in the end. Would you rather I use `findMonitorForActor` or not? I can test if it works for instance.